### PR TITLE
Lightweight `testlib`

### PR DESCRIPTION
**Pull request by nikurt**
Remove the dependency of `testlib` on `nearcore` by extracting tests and code dependeing on `nearcore` into `integration-tests`.

### CODE REVIEW

**Review by matklad — APPROVED** [review]
Don't have capacity for a super thorough review at this point, but this moves into the right direction!

File: test-utils/run-nodes/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "run-nodes"
+version = "0.1.0"
+authors = ["Near Inc <hello@nearprotocol.com>"]
+edition = "2018"
+
+[[bin]]
+name = "run-nodes"
```
[the remark this answers is not in the document]
Comment (nikurt) [reply test-utils/run-nodes/Cargo.toml:8]: Don't know. Somebody can build and run it manually. Maybe somebody has a bash script that runs this ¯\_(ツ)_/¯.
I suggest to keep for now and delete later when proven unused

**Review by frol — APPROVED** [review]
I see many things move around. The whole reorg makes total sense to me, but I fear I could miss some minor details among the changes; are there any?

Also, can we measure the wins that we achieve with this reorg (e.g. compilation / testing times)? I would love to see those

**Comment by frol** [comment]
Just to bring some color here, @miraclx is also working on quality of life improvements for nearcore codebase, and here is a recent win: https://github.com/near/nearcore/pull/4651#issuecomment-895578948

@nikurt Thanks for pulling this out!

File: test-utils/run-nodes/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (bowenwang1996) [reply test-utils/run-nodes/Cargo.toml:8]: I don't think so. This looks like some legacy code that should be removed

**Review by bowenwang1996 — APPROVED** [review]

**Comment by nikurt** [comment]
> Just to bring some color here, @miraclx is also working on quality of life improvements for nearcore codebase, and here is a recent win: [#4651 (comment)](https://github.com/near/nearcore/pull/4651#issuecomment-895578948)
Thanks for the link. Now I'm also curios, will measure the impact and report here.

**Comment by nikurt** [comment]
> minor details among the changes; are there any?
I tried to keep the PR as simple as possible, without making any changes unrelated to the refactoring. The worst that could happen is an artifact of merging, which hopefully is caught by buildkite.

**Comment by nikurt** [comment]
Created #4670 to remove `run_nodes.rs`.

**Comment by nikurt** [comment]
> > Just to bring some color here, @miraclx is also working on quality of life improvements for nearcore codebase, and here is a recent win: [#4651 (comment)](https://github.com/near/nearcore/pull/4651#issuecomment-895578948)
> > Thanks for the link. Now I'm also curios, will measure the impact and report here.

`cargo test -p node-runtime`
debug clean build time: 4m 30s / 4m 20s
debug binary size: 78 MB / 71 MB
dependencies: 612 / 591
debug test runtime:  6m 0s / 30s

`cargo test -p nearcore`
debug clean build time: 4m 46s / 4m 9s
debug binary size: 96M / 96M
dependencies: 616 / 608
debug test runtime: 9m 50s / 5s

**Pull request merged**